Mathlib/Order/RelIso/Set.lean

```diff
@@ -132,3 +132,83 @@ theorem wellFounded_iff_wellFounded_subrel {r : α → α → Prop} [IsTrans α
     WellFounded r ↔ ∀ b, WellFounded (Subrel r { a | r a b }) where
   mp h _ := InvImage.wf Subtype.val h
   mpr h := ⟨fun a ↦ ⟨_, fun b hr ↦ ((h a).apply _).of_subrel ⟨b, hr⟩⟩⟩
+
+namespace Equiv
+
+variable {r : α → α → Prop} {x : α} [IsTrans α r] [IsTrichotomous α r] [DecidableRel r]
+
+variable (r x) in
+/-- A relation is isomorphic to the lexicographic sum of elements less than `x` and elements not
+less than `x`. -/
+def sumLexComplLeft : Sum.Lex (Subrel r {y | r y x}) (Subrel r {y | ¬ r y x}) ≃r r := by
+  refine ⟨Equiv.sumCompl _, ?_⟩
```

Comment:
Since this is data, can you use `where` notation to build this instead of `by`?

Reply:
That would require me to specify five fields, whereas by using constructor notation directly I can immediately take care of four of them. The last field, `map_rel_iff'`, is just the remainder of the proof.

Reply:
Isn't `refine` generally fine to use with data? I think what I'm writing should be equivalent to
```lean 4
⟨Equiv.sumCompl _, by
  long_proof
⟩
```

Reply:
> That would require me to specify five fields,

This is not true, as Yael remarks below

> Isn't `refine` generally fine to use with data?

In practice yes; stylistically, no - your "equivalent to" version is preferred (and Yael's version even more so)